source/s3_client.c

```diff
@@ -149,6 +150,19 @@ void aws_s3_set_dns_ttl(size_t ttl) {
     s_dns_host_address_ttl_seconds = ttl;
 }
 
+/**
+ * Determine how many connections are ideal by dividing target-throughput by throughput-per-connection.
+ * TODO: we may consider to alter this, upload to regular s3 can use more connections, and s3 express
+ * can use less connections to reach the target throughput..
+ **/
+static uint32_t s_get_ideal_connection_number_from_throughput(double throughput_gps) {
+    double ideal_connection_count_double = throughput_gps / s_throughput_per_connection_gbps;
+    /* round up and clamp */
+    ideal_connection_count_double = ceil(ideal_connection_count_double);
+    ideal_connection_count_double = aws_min_double(UINT32_MAX, ideal_connection_count_double);
+    return (uint32_t)ideal_connection_count_double;
+}
+
 /* Returns the max number of connections allowed.
  *
  * When meta request is NULL, this will return the overall allowed number of connections.
```

Comment:
if you expand the doc, this is no longer accurate right? it will now return the number of connections that can be made for that specific meta request and not just meta request type